archinstall/lib/installer.py

```diff
@@ -888,19 +888,48 @@ def add_grub_bootloader(self, boot_partition :Partition, root_partition :Partiti
 
 		root_fs_type = get_mount_fs_type(root_partition.filesystem)
 
+		_file = "/etc/default/grub"
+
+		try:
+			with open(_file, 'r') as fh:
+				contents = fh.readlines()
+		except FileNotFoundError:
+			log(f"Could not configure GRUB, file not found: '{_file}'.", level=logging.DEBUG)
+			return False
+
+		options = [f'rootfstype={root_fs_type}']
+
+		if self._zram_enabled:
+			options.insert(0, 'zswap.enabled=0')
+
 		if real_device := self.detect_encryption(root_partition):
 			root_uuid = SysCommand(f"blkid -s UUID -o value {real_device.path}").decode().rstrip()
-			_file = "/etc/default/grub"
-			add_to_CMDLINE_LINUX = f"sed -i 's/GRUB_CMDLINE_LINUX=\"\"/GRUB_CMDLINE_LINUX=\"cryptdevice=UUID={root_uuid}:cryptlvm rootfstype={root_fs_type}\"/'"
-			enable_CRYPTODISK = "sed -i 's/#GRUB_ENABLE_CRYPTODISK=y/GRUB_ENABLE_CRYPTODISK=y/'"
```

Comment:
I haven't used GRUB in a millennia. But I do agree that explicitly creating a config file is probably better.
But are there other options in the default conf that we would lose if we don't use the `sed` approach?

Way cleaner code the way this PR suggests, just checking in if we're losing anything by doing it?

Reply:
I also do not use GRUB. I did a minimal amount of testing on the code using the `grub.default` file, which is distrusted as `/etc/default/grub`, from the package source for GRUB on Arch Linux.

This should produces an identical copy of the `/etc/default/grub` file as the `sed` approach does and is just the pythonic way of doing it.